e2e/upgrade-cephfs.go

```diff
@@ -157,6 +210,122 @@ var _ = Describe("CephFS Upgrade Testing", func() {
 				}
 			})
 
+			By("Create clone from a snapshot", func() {
+				pvcClonePath := cephfsExamplePath + "pvc-restore.yaml"
+				appClonePath := cephfsExamplePath + "pod-restore.yaml"
+				v, err = f.ClientSet.Discovery().ServerVersion()
+				if err != nil {
+					e2elog.Logf("failed to get server version with error %v", err)
+					Fail(err.Error())
+				}
+				// pvc clone is only supported from v1.16+
+				if v.Major > "1" || (v.Major == "1" && v.Minor >= "17") {
+					pvcClone, err = loadPVC(pvcClonePath)
+					if err != nil {
+						e2elog.Failf("failed to load pvc with error %v", err)
+					}
+					pvcClone.Namespace = f.UniqueName
+					pvcClone.Spec.Resources.Requests[v1.ResourceStorage] = resource.MustParse(pvcSize)
+					appClone, err = loadApp(appClonePath)
+					if err != nil {
+						e2elog.Failf("failed to load application with error %v", err)
+					}
+					appClone.Namespace = f.UniqueName
+					appClone.Name = "snap-clone-cephfs"
+					appClone.Labels = map[string]string{"app": "validate-snap-cephfs"}
+					err = createPVCAndApp("", f, pvcClone, appClone, deployTimeout)
+					if err != nil {
+						e2elog.Failf("failed to create pvc and application with error %v", err)
+					}
+					opt := metav1.ListOptions{
+						LabelSelector: "app=validate-snap-cephfs",
+					}
+					mountPath := getMountPath(appClone)
+					testFilePath := filepath.Join(mountPath, "testClone")
+
+					newCheckSum, err = calculateSHA512sum(f, appClone, testFilePath, &opt)
+					if err != nil {
+						e2elog.Failf("failed to calculate checksum of %s", testFilePath)
+					}
+
+					if !strings.Contains(newCheckSum, checkSum) {
+						e2elog.Failf("The md5sum of files did not match, expected %s received %s  ", checkSum, newCheckSum)
```

Comment:
not `md5sum` anymore, just refer to it as checksum like you do almost everywhere